learning_journal/views/views.py

```diff
@@ -0,0 +1,45 @@
+from pyramid.view import view_config
+from pyramid.response import Response
+from sqlalchemy.exc import DBAPIError
+from ..models import Entry
+
+DB_ERROR = "Whoops, there was a problem with the database!"
+
+
+@view_config(route_name='index', renderer='../templates/index.jinja2')
+def index_view(request):
+    try:
+        all_entries = request.dbsession.query(Entry).order_by(Entry.id.desc())
+    except DBAPIError:
+        return Response(DB_ERROR, content_type="text/plain", status=500)
```

Comment:
Once you've gone to production, I'd consider allowing DBAPI errors to pass unhandled.  The framework is going to send a 500 response anyway.  Do you need to advertise to your visitor that there was a database connectivity issue?